apps/backend/src/users/users.service.ts

```diff
@@ -37,6 +38,40 @@ export class UsersService {
     return await this.prisma.user.findMany({ omit: { weight: true } });
   }
 
+  async calBac(authSchId: string): Promise<UserBac> {
+    const user = await this.prisma.user.findUnique({
+      where: { authSchId },
+      select: { weight: true, gender: true },
+    });
+
+    if (!user) {
+      throw new NotFoundException('User not found');
+    }
+
+    const currentTime = new Date();
+    const userWeightInGrams = user.weight * 1000;
+    const genderFactor = user.gender === 'Male' ? 0.68 : 0.55;
+
+    const drinkActions = await this.prisma.drinkAction.findMany({
+      // where: { authSchId },
+      include: { drink: { select: { alcoholContent: true } } },
+    });
```

Comment:
it would probably make sense to exclude drink actions that happened more than X (let's say 12) hours ago. As the app gets more popular, the users can have loads of drinkactions, but only the most recents will affect the BAC, so we shouldn't waste time calculating the older ones